codegen/core/src/main/java/software/amazon/smithy/python/codegen/ClientGenerator.java

```diff
@@ -553,6 +554,14 @@ await sleep(retry_token.retry_delay)
                                     )
                                 )
                                 logger.debug("Signed HTTP request: %s", context.transport_request)
+
+                                # TODO - Move this to separate resolution/population function
+                                fields = context.transport_request.fields
+                                auth_value = fields["Authorization"].as_string()  # type: ignore
+                                signature = re.split("Signature=", auth_value)[-1]  # type: ignore
+                                context.properties["signature"] = signature.encode('utf-8')
```

Comment:
The sigv4 specific bits here we can add a code interceptor that appends to the SignRequestSection into the AwsAuthIntegration.

Though I'm not sure how this fits into the pure python request pipeline.